### DISCUSSION TITLE
Converting `List[Output[T]]` to `Output[List[T]]`

### DISCUSSION THREAD

**Discussion by lovesegfault** [pulumi/pulumi#9340] [Pulumi in Practice] [answered]
I'm using the Python SDK, and I end up holding a `List[Output[SomeData]]`, which I want to process as a `List[SomeData]`. I figured the correct way of doing this would be to "lift" the `Output` to encompass the whole list, i.e. having an `Output[List[SomeData]]`. I did this:

```python3
# original comes from somewhere else
original: List[pulumi.Output[SomeData]]
transposed: pulumi.Output[List[SomeData]]
transposed = pulumi.Output.from_input([])
for data in original:
    transposed = transposed.apply(
        lambda acc, data=data: data.apply(lambda curr: acc + [curr])
    )
```

This works OK, but makes mypy stop working because it gives up on the gnarly nested apply.

Is there a better way to do this? I feel like `pulumi.Output.all` should be able to help me here, but I can't seem to figure out how to hold it the right way.

Cheers!

**Comment by Frassle** [comment, marked as answer]
> I feel like pulumi.Output.all should be able to help me here, but I can't seem to figure out how to hold it the right way.

`all` is defined as `all(*args: Input[T]) -> "Output[List[T]]"` so you want to call it with the unpack operator if you've already got an array, e.g: `Output.all(*array_of_outputs)`

**Reply by lovesegfault** [reply]
Ah, bingo, I had incorrectly assumed that all was `all(args: Sequence[Input[T]])` and was trying to pass in `my_list` instead of `*my_list`.

Thanks!
